src/app/pages/client/explore/Explore.tsx

```diff
@@ -93,79 +109,181 @@ export function AddServer() {
                 variant="Surface"
                 size="500"
               >
-                <Box grow="Yes">
-                  <Text size="H4">Add Server</Text>
-                </Box>
+                <Box grow="Yes">{header}</Box>
                 <IconButton size="300" onClick={() => setDialog(false)} radii="300">
                   <Icon src={Icons.Cross} />
                 </IconButton>
               </Header>
-              <Box
-                as="form"
-                onSubmit={handleSubmit}
-                style={{ padding: config.space.S400 }}
-                direction="Column"
-                gap="400"
-              >
+              <Box style={{ padding: config.space.S400 }} direction="Column" gap="400">
                 <Text priority="400">Add server name to explore public communities.</Text>
                 <Box direction="Column" gap="100">
                   <Text size="L400">Server Name</Text>
                   <Input ref={serverInputRef} name="serverInput" variant="Background" required />
-                  {exploreState.status === AsyncStatus.Error && (
+                  {viewState.status === AsyncStatus.Error && (
                     <Text style={{ color: color.Critical.Main }} size="T300">
                       Failed to load public rooms. Please try again.
                     </Text>
                   )}
                 </Box>
                 <Box direction="Column" gap="200">
-                  {/* <Button
+                  <Button
                     type="submit"
+                    onClick={handleView}
                     variant="Secondary"
+                    fill="Soft"
                     before={
-                      exploreState.status === AsyncStatus.Loading ? (
+                      viewState.status === AsyncStatus.Loading && (
                         <Spinner fill="Solid" variant="Secondary" size="200" />
```

Comment:
button has "Soft" fill but spinner has "Solid". it should have same fill as button it is in.